### PR TITLE
networkAware update configs

### LINKED ISSUE #461: [Network Aware scheduling] The config and deployment files need to be updated
I noticed the config and deployment files need to be updated. Also, can I add the Network-aware scheduling plugins to the main README file? :)  

I will be working on these modifications. 
/assign

### PR DESCRIPTION
**Pull request by jpedro1992**
#### What type of PR is this?

Discuss and solve issue #461 about the config and deployment files of the networkAware plugins. 

#### What this PR does / why we need it:

The deployment files included in the manifest folder have been updated. Also, the networkAware plugins have been made visible in the main README file.  

#### Which issue(s) this PR fixes:
Fixes #461

### CODE REVIEW

File: manifests/networktopology/deploy.yaml [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+  template:
+    metadata:
+      labels:
+        component: scheduler
+        tier: control-plane
+        version: second
+    spec:
+      nodeSelector:
+        node-role.kubernetes.io/master: ""
+      serviceAccountName: network-aware-scheduler
```
Comment (ttsokov) [inline manifests/networktopology/deploy.yaml:24]: The following rules have to be added into the ClusterRole, which is bound to the ServiceAccount with name `network-aware-scheduler`:
```
- apiGroups: ["appgroup.diktyo.k8s.io"]
  resources: ["appgroups"]
  verbs: ["get", "list", "watch", "create", "delete", "update", "patch"]
- apiGroups: ["networktopology.diktyo.k8s.io"]
  resources: ["networktopologies"]
  verbs: ["get", "list", "watch", "create", "delete", "update", "patch"]
```
Probably RBAC yaml file containing the ServiceAccount, ClusterRole, ClusterRoleBinding, Role and RoleBinding can be added inside `manifests/networktopology/` directory, similar to this one - https://github.com/kubernetes-sigs/scheduler-plugins/blob/master/manifests/noderesourcetopology/cluster-role.yaml?

File: manifests/appgroup/deploy.yaml [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+  template:
+    metadata:
+      labels:
+        component: scheduler
+        tier: control-plane
+        version: second
+    spec:
+      nodeSelector:
+        node-role.kubernetes.io/master: ""
+      serviceAccountName: network-aware-scheduler
```
Comment (ttsokov) [inline manifests/appgroup/deploy.yaml:24]: The following rules have to be added into the ClusterRole, which is bound to the ServiceAccount with name `network-aware-scheduler`:
```
- apiGroups: ["appgroup.diktyo.k8s.io"]
  resources: ["appgroups"]
  verbs: ["get", "list", "watch", "create", "delete", "update", "patch"]
- apiGroups: ["networktopology.diktyo.k8s.io"]
  resources: ["networktopologies"]
  verbs: ["get", "list", "watch", "create", "delete", "update", "patch"]
```
Probably RBAC yaml file containing the ServiceAccount, ClusterRole, ClusterRoleBinding, Role and RoleBinding can be added inside `manifests/appgroup/` directory, similar to this one - https://github.com/kubernetes-sigs/scheduler-plugins/blob/master/manifests/noderesourcetopology/cluster-role.yaml?

Comment (jpedro1992) [reply manifests/appgroup/deploy.yaml:24]: Thank you for suggestion @ttsokov! I will update the PR accordingly.

**Review by Huang-Wei — COMMENTED** [review]
Thanks @jpedro1992 . Could you also add the RBAC bits to manifests/install/charts/as-a-second-scheduler/templates/rbac.yaml, and then modify manifests/install/charts/as-a-second-scheduler/values.yaml locally (don't need to check in) to see if Helm install and your plugin function well?

**Comment by jpedro1992** [comment]
Thanks @Huang-Wei for the review. I will update the PR.

**Comment by jpedro1992** [comment]
Hi @Huang-Wei and @ttsokov, 

Thank you for your review! I added a couple of deployment files (appgroup-controller, networktopology-controller, etc) and a few comments to a few files in order to properly deploy the scheduler with network-aware plugins. 

A few notes below: 

- Building the image locally and running it on K8s v1.23.X does not work. I face the following error: 

`"E0130 21:35:32.938159       1 reflector.go:138] [k8s.io/client-go/informers/factory.go:134](http://k8s.io/client-go/informers/factory.go:134): Failed to watch *v1.CSIStorageCapacity: failed to list *v1.CSIStorageCapacity: the server could not find the requested resource".` 

All pods deployed with the scheduler are stuck in a pending state. 

- Building the image locally or using the registry.k8s.io/scheduler-plugins/kube-scheduler:v0.24.9 and running it on K8s v1.24.4 works. I added a comment mentioning that network aware plugins should be used with scheduler plugins v.0.24.9

**Comment by Huang-Wei** [comment]
> Building the image locally and running it on K8s v1.23.X does not work

That's not surprising given you're running a high-version scheduler (for master it's 1.25) against to a low-version API Server (1.23).

**Comment by Huang-Wei** [comment]
/hold

@jpedro1992 could you take a look at the CI failure?

**Comment by jpedro1992** [comment]
Hi @Huang-Wei, I rebased the PR and everything seems fine now. For some reason, it was not assuming the previous commit that had solved the problem with the integration test.

**Pull request merged**